core/src/main/java/tc/oc/pgm/namedecorations/NameDecorationRegistryImpl.java

```diff
@@ -71,6 +72,12 @@ public void onPartyChange(PlayerPartyChangeEvent event) {
     player.setDisplayName(getDecoratedName(player, party == null ? null : party.getColor()));
   }
 
+  @EventHandler
```

Comment:
You may want this on monitor, so that the leave message can make use of cached name